episodes/filters-and-thresholding.md

```diff
@@ -0,0 +1,743 @@
+seems much smoother and has fewer areas inside nuclei that aren't labelled, as 
+well as fewer areas of the background that are incorrectly labelled. 
+
+This workflow of blurring an image then thresholding it is a very common method 
+to provide a smoother, more complete mask. Note that it still isn't perfect! If 
+you browse through the z slices with the slider at the bottom of the image, it's 
+clear that some areas are still missed or incorrectly labelled.
+
+## Automated thresholding
+
+First, let's clean up our layer list again. Make sure you only have the 'nuclei' 
+and 'Result of gaussian_blur' layers in the layer list - select any others and 
+remove them by clicking the ![](
+https://raw.githubusercontent.com/napari/napari/main/napari/resources/icons/delete.svg
+){alt="A screenshot of Napari's delete layer button" height='30px'} icon. Then, 
+if you still have the `napari-matplotlib` histogram open, close it by clicking 
+the tiny `x` icon in the top left corner.
+
+So far we have chosen our threshold manually by looking at the image histogram, 
+but it would be better to find an automated method to do this for this. Many 
+such methods exist - for example, see the various options starting with 
+'Threshold' under `Tools > Segmentation / binarization`. One of the most common 
+methods is _Otsu thresholding_, which we will look at now.
+
+Let's go ahead and apply this to our blurred image:  
+`Tools > Segmentation / binarization > Threshold (Otsu et al 1979, scikit-image, nsbatwm)`
+
+This will open a panel on the right side of Napari. Select 'Result of 
+gaussian_blur(data)' in the image row, then click Run:
+
+![](fig/threshold-blurred-otsu-mask.png){alt="Mask of nuclei (brown) overlaid on 
+nuclei image - created with Otsu thresholding after gaussian blur"}
+
+This should produce a mask that is almost identical to the one we created with a 
+manual threshold.
+
+How is Otsu finding the correct threshold? The details are rather complex, but 
+to summarise - Otsu's method is trying to find a threshold that minimises the 
+spread of pixel values in both the background and your class of interest. In 
+effect, it's trying to find two peaks in the image histogram and place a 
+threshold in between them.
+
+This means it's only suitable for images that show two clear peaks in their 
+histogram (also known as a 'bimodal' histogram). Other images may require 
+different automated thresholding methods to produce a good result. 
+
+::::::::::::::::::::::::::::::::::::: challenge 
+
+## Automated thresholding
+
+For this exercise, we'll use the same example image as the [manual thresholding 
+exercise](#manual-thresholds). If you don't have that image open, run the top 
+code block in that exercise to open the image:
+
+![](fig/manual-thresholding-exercise-shapes.png){alt="Noisy image containing a 
+rectangle, circle and triangle"}
+
+Try some of the other automatic thresholding options provided by the 
```

Comment:
For some of these it would be nice if we could somehow set the colors differently so that it is easier to compare masks